ebuild-writing/bundled-deps/text.xml

```diff
@@ -0,0 +1,390 @@
+    bug 471682</uri>, <uri link="https://bugs.php.net/bug.php?id=66095">
+    PHP bug 66095</uri>).
+  </li>
+</ul>
+</body>
+</subsection>
+</section>
+
+<section>
+<title>Downstream consequences</title>
+<body>
+
+<p>
+When a bundled dependency is discovered downstream this has a number of
+bad consequences.
+</p>
+
+</body>
+
+<subsection>
+<title>Analysis</title>
+<body>
+
+<p>
+So there is a copy of libvorbis bundled with that media player. Which
+version is it? Has it been modified?
+</p>
+</body>
+
+<subsubsection>
+<title>Separating forks from copies</title>
+<body>
+
+<p>
+Before the bundled dependency can be replaced by the system-widely installed
+one, we need to know if it has been modified: we have to know if it's a fork.
+</p>
+
+<p>
+If it is a fork it may or may not be replaced without breaking something.
+</p>
+
+<p>
+That's something to find out: more time wasted. If the code says which
+version it is we at least know what to run <c>diff</c> against, but that
+is not always the case.
+</p>
+</body>
+</subsubsection>
+
+<subsubsection>
+<title>Determining versions</title>
+<body>
+
+<p>
+If a bundled dependency doesn't tell its version we may have to find out
+ourselves. Mailing upstream could work, comparing against a number of
+tarball contents may work too. Lots of opportunities to waste time.
+</p>
+</body>
+</subsubsection>
+</subsection>
+
+<subsection>
+<title>Patching</title>
+<body>
+
+<p>
+Once it is clear that a bundled dependency can be ripped out, a patch is
+written, applied and tested (more waste of time). If upstream is willing to
+co-operate the patch may be dropped later. If not the patch will need
+porting to each now version downstream.
+</p>
+</body>
+</subsection>
+
+<subsection>
+<title>What to do upstream</title>
+<body>
+
+<ul>
+  <li>
+    <p>
+    Remove bundled dependency:
+    </p>
+
+    <p>
+    At best, remove the bundle dependency and allow
+    compilation against dependency <e>D</e> from either a system-wide installation
+    of it or a local one at any user-defined location.
+    </p>
+
+    <p>
+    That gives flexibility to users on systems without <e>D</e> packaged and makes
+    it easy to compile against the system copy downstream: cool!
+    </p>
+  </li>
+  <li>
+    <p>
+    Keep bundled dependency: make usage <e>completely optional</e>:
+    </p>
+
+    <p>
+    With a build time option to disable use of the bundled dependency it is
+    possible to bypass it downstream without patching: nice!
+    </p>
+
+    <p>
+    When keeping dependency <e>D</e> bundled make sure to follow the upstream of
+    <e>D</e> closely and update your copy to a recent version of <e>D</e> on every
+    minor (and major) release to at least reduce the damage done to people
+    using your bundled version a little.
+    </p>
+
+    <p>
+    Also: Clearly document if a bundled dependency is a fork or an unmodified
+    copy and which version of the bundled software we are dealing with.
+    </p>
+  </li>
```

Comment:
The blank lines before the `<p>` tags are not needed within list items. (Sorry, I had missed this in the first round.)